src/components/Provider/BreakpointProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import debounce from 'lodash/debounce';
+import * as React from 'react';
+import {
+	Breakpoint,
+	BreakpointContext,
+} from '../../contexts/BreakpointContext';
+export interface BreakpointProviderProps {
+	breakpoints: number[];
+	onBreakpointChange?: (breakpoint: Breakpoint) => void;
+}
+
+export const BreakpointProvider: React.FC<BreakpointProviderProps> = ({
+	breakpoints,
+	children,
+}) => {
+	const [currentBreakpoint, setCurrentBreakpoint] = React.useState(
+		0 as Breakpoint,
+	);
+
+	const handleResize = () => {
+		const clientWidth = window.innerWidth;
+		if (clientWidth < breakpoints[0]) {
```

Comment:
I am thinking, we probably want this to be more dynamic, as the length of the `breakpoints` array can change, and the hook should be able to handle it seamlessly.